### ISSUE TITLE
mysql_replication: master_use_gtid

### ISSUE THREAD

**Issue by wedi** [ansible-collections/ansible.mysql#11]
##### SUMMARY
I am using a recent version of ansible and a feature [merged back in 2019](https://github.com/ansible/ansible/pull/62648) is not available although the code is still here: https://github.com/ansible-collections/community.mysql/blob/c255d7ece29d79e57d17ad0326a0d6da88c64139/plugins/modules/mysql_replication.py#L109-L118

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
mysql_replication

##### ANSIBLE VERSION

```paste below
ansible --version
ansible 2.9.10
  config file = /Users/weise/code/server-provisioning/ansible.cfg
  configured module search path = ['/Users/weise/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.9.10/libexec/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 21 2020, 10:42:08) [Clang 11.0.0 (clang-1100.0.33.17)]

```

##### CONFIGURATION

```shell
$ ansible-config dump --only-changed
ANSIBLE_NOCOWS(env: ANSIBLE_NOCOWS) = True
```

##### OS / ENVIRONMENT

Mac OS X

##### STEPS TO REPRODUCE

```yaml
- mysql_replication:
    mode: changemaster
    master_user: "{{ mariadb__replication_user.username }}"
    master_host: "{{ mariadb__replication_user.host }}"
    master_password: "{{ mariadb__replication_user.password }}"
    master_use_gtid: current_pos
    login_unix_socket: /run/mysqld/mysqld.sock
```

##### EXPECTED RESULTS

I'd expect that `CHANGE MASTER` gets sucessfully executed on the database.

##### ACTUAL RESULTS

it fails. I get a complaint about a parameter not being available that was [merged back in October 2019](https://github.com/ansible/ansible/pull/62648). This [code is available in this repo](https://github.com/ansible-collections/community.mysql/blob/c255d7ece29d79e57d17ad0326a0d6da88c64139/plugins/modules/mysql_replication.py#L109) but it looks like it's not included in the latest ansible release.

https://github.com/ansible-collections/community.mysql/blob/c255d7ece29d79e57d17ad0326a0d6da88c64139/plugins/modules/mysql_replication.py#L109-L118

```paste below
Unsupported parameters for (mysql_replication) module: master_use_gtid Supported parameters include: ca_cert, client_cert, client_key, config_file, connect_timeout, login_host, login_password, login_port, login_unix_socket, login_user, master_auto_position, master_connect_retry, master_host, master_log_file, master_log_pos, master_password, master_port, master_ssl, master_ssl_ca, master_ssl_capath, master_ssl_cert, master_ssl_cipher, master_ssl_key, master_user, mode, relay_log_file, relay_log_pos
```

**Comment by bmildren** [comment]
Hi @wedi 

This looks like it was a new feature introduced post 2.9 (https://github.com/ansible/ansible/blob/stable-2.9/lib/ansible/modules/database/mysql/mysql_replication.py), as I understand it, new features aren't backported.  That said, this is partly the motivation in switching to collections, so the release cycle of the modules can be decoupled from `ansible-base` (decreasing the time it takes for module changes to be included in the ansible release).

This feature and others will be included in Ansible from 2.10 (Ansible Community Distribution), however in the meantime you can include the module/collection locally as per: https://docs.ansible.com/ansible/latest/dev_guide/developing_locally.html#adding-a-module-locally.  (following this, as an interim measure the local module can override the module included in 2.9 until 2.10 is available).

**Comment by Andersson007** [comment]
Hi, yes, @bmildren is right, backporting features is forbidden in the project.
Though community.mysql collection 1.0.0 has recently been released https://galaxy.ansible.com/community/mysql
@wedi you could download and try https://docs.ansible.com/ansible/latest/user_guide/collections_using.html.
It should work with 2.9 as far as i know.
Any feedback would be much valuable. 
So, i'm closing the issue. Thanks!
